platform/darwin/src/MGLStyleValue.h

```diff
@@ -135,77 +172,315 @@ MGL_EXPORT
+/**
+ Creates and returns an `MGLStyleFunction` object representing a camera function. 
+ If the function is set as a style layer property value, it will be interpreted 
+ as a camera function with a linear interpolation curve.
 
-// TODO: API docs
+ @note Do not create instances of `MGLStyleFunction` unless it is required for
+    backwards compatiblity with your application code. You should create and use
+    instances of `MGLCameraStyleFunction` to specify how properties will
+    be visualized at different zoom levels.
+
+ @param stops A dictionary associating zoom levels with style values.
+ @return An `MGLStyleFunction` object with the given stops.
+ */
++ (instancetype)functionWithStops:(NS_DICTIONARY_OF(NSNumber *, MGLStyleValue<T> *) *)stops __attribute__((deprecated("Use +[MGLStyleValue valueWithInterpolationMode:cameraStops:options:]")));
+
+/**
+ Creates and returns an `MGLStyleFunction` object representing a camera function.
+ If the function is set as a style layer property value, it will be interpreted 
+ as a camera function with an interpolation curve controlled by the provided 
+ interpolation base.
+
+ @note Do not create instances of `MGLStyleFunction` unless it is required for
+    backwards compatiblity with your application code. You should create and use
+    instances of `MGLCameraStyleFunction` to specify how properties will
+    be visualized at different zoom levels.
+
+ @param interpolationBase The exponential base of the interpolation curve.
+ @param stops A dictionary associating zoom levels with style values.
+ @return An `MGLStyleFunction` object with the given interpolation base and stops.
+ */
++ (instancetype)functionWithInterpolationBase:(CGFloat)interpolationBase stops:(NS_DICTIONARY_OF(NSNumber *, MGLStyleValue<T> *) *)stops __attribute__((deprecated("Use +[MGLStyleValue valueWithInterpolationMode:cameraStops:options:]")));
+
+#pragma mark Initializing a Style Function
+
+/**
+ Returns an `MGLStyleFunction` object representing a camera function. If the 
+ function is set as a style layer property value, it will be interpreted
+ as a camera function with an interpolation curve controlled by the provided
+ interpolation base.
+ 
+ @note Do not create instances of `MGLStyleFunction` unless it is required for
+    backwards compatiblity with your application code. You should create and use
+    instances of `MGLCameraStyleFunction` to specify how properties will
+    be visualized at different zoom levels.
+
+ @param interpolationBase The exponential base of the interpolation curve.
+ @param stops A dictionary associating zoom levels with style values.
+ @return An `MGLStyleFunction` object with the given interpolation base and stops.
+ */
+- (instancetype)initWithInterpolationBase:(CGFloat)interpolationBase stops:(NS_DICTIONARY_OF(NSNumber *, MGLStyleValue<T> *) *)stops NS_DESIGNATED_INITIALIZER;
```

Comment:
> If you mark this as a designated initializer, does that mean all the subclass initializers have to call this method? 

Yes

Or would removing NS_DESIGNATED_INITIALIZER be considered a breaking change?

I don't think so. We never suggested that developers should subclass MGLStyleFunction.

In 7cbeba7, I exposed only MGLStyleFunction's initializer publicly but
deprecated it and removed its designated initializer status. Internally this allows us to ignore the deprecated initializer in subclasses.